### PR TITLE
New message indicator

### PR DESCRIPTION
**Pull request by gabriel**
Adds a new message indicator.

2016-11-18 at 4 09 pm

- When messages are appended to a conversation, it marks the first new message.
- When new messages are appended:
  - If conversation not visible, if new message is not set, set it  
  - If conversation is visible, clear new message (if one is set)
- When you select a different conversation it clears the first new message marker of the conversation you were just on (you've seen the messages). I take care to only do this when a user selects a conversation through click action (if the conversation select was programmatic it won't clear the indicator).

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/react-hackers

**Comment by chrisnojima** [comment]
Can we add a TODO that the logic for the currently selected message. If you get a message on the currently selected conversation and the electron window isn't focused then we actually do get the orange line. This isn't MVP but is something we'll want later.

File: shared/chat/conversation/messages/text.desktop.js [hunk 1]
```diff
@@ -56,4 +56,8 @@ const MessageTextWrapper = ({author, message, messageState, followState, style,
   </Box>
 )
 
+const stylesFirstNewMessage = {
+  borderTop: 'solid 1px #ff6f21',
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/text.desktop.js]:
```
`solid 1px ${globalColors.orange}`
```

**Comment by chrisnojima** [comment]
Doesn't seem like the currently selected logic is working for me. If i have a conversation selected and i get a new message it draws the orange line anyways

**Comment by keybase-ci-visdiff** [comment]
The commits 375194da062c...e034f82611b5 introduce visual changes on linux. <details><summary>:mag_right: 3 changed</summary>

 * Changed: **unlock_folders-Single** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-Single-375194da062c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-Single-e034f82611b5.png)  
   unlock_folders-Single comparison
 * Changed: **unlock_folders-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-Normal-375194da062c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-Normal-e034f82611b5.png)  
   unlock_folders-Normal comparison
 * Changed: **unlock_folders-No_paperkeys** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-No_paperkeys-375194da062c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-No_paperkeys-e034f82611b5.png)  
   unlock_folders-No_paperkeys comparison
</summary>

**Comment by gabriel** [comment]
Ok fixed the issue where new message wasn't clearing indicator... I updated the PR description rules to reflex this. Also added some more comments in code.

**Pull request merged**